### DISCUSSION TITLE
Big FPS drop during PrimeTweenManager.Update()

### DISCUSSION THREAD

**Discussion by Alaxxxx** [KyryloKuzyk/PrimeTween#42] [FAQ] [answered]
Hi,

I have a question related to this fps drop, I can't figure out what is this File.Read coming from PrimeTween
![SharedScreenshot](https://github.com/KyryloKuzyk/PrimeTween/assets/130758363/71e8daef-84cf-4a3a-9d8b-7950bd7f306a)

I tried in a release build and no fps drop, only in the editor.  So no big deal as it will surely have no impact on release but I'm still curious about this.
Edit: Forgot to mention that this happen during a simple falling animation where I use Tween.Custom() with no callback and this is an android build.

Have an idea ?

**Comment by KyryloKuzyk** [comment, marked as answer]
This spike is most likely coming from a `Tween.Delay()` or from `.OnComplete()` callback that calls `Loading.ReadObject` method. `Loading.ReadObject` is not part of PrimeTween, PrimeTween is only calling another method from your codebase which causes the spike.
Please enable 'Deep profiling' to see the exact place in your codebase that makes this call or search your codebase to find out which places are making this call.

**Reply by KyryloKuzyk** [reply]
@OpGala My assumption is that you have some kind of loading screen, and you're loading the level in the OnComplete() callback. And because OnComplete() is called by PrimeTween, you see the spike as also coming from PrimeTween.

**Reply by Alaxxxx** [reply]
So you were absolutely right. The peak was triggered by a script running in the background that I'd forgotten about. It loaded an interface window that didn't appear immediately but was loaded and executed anyway.
And indeed I was using PrimeTween with callbacks that called a script that for some reason continuously loads textures hence the "Read.File" it applies.
As it was a different UI system that I'm still getting to grips with called "Nova UI", which is excellent by the way, I had no idea it could have come from there as I'm actually converting Unity UI to this one in the same time I converted to PrimeTween from DoTween.

But thanks to you, sorry for the mistake, I love your work, keep it up!

**Reply by KyryloKuzyk** [reply]
@OpGala I'm glad that helped! Please consider leaving an honest review on Asset Store, it helps a lot :)

**Reply by Alaxxxx** [reply]
I already left a review 2 months ago with the same pseudo than here that's the least I could have done
